src/core/index.ts

```diff
@@ -93,57 +101,117 @@ export interface PromptUiActions {
     };
 }
 
+export interface UnderlayConfig {
+    colors: {
+        bg: string;
+    };
+}
+
+export enum PromptPosition {
+    TOP_LEFT = 'top-left',
+    TOP_CENTER = 'top-center',
+    TOP_RIGHT = 'top-right',
+    CENTER_LEFT = 'center-left',
+    CENTER = 'center',
+    CENTER_RIGHT = 'center-right',
+    BOTTOM_LEFT = 'bottom-left',
+    BOTTOM_CENTER = 'bottom-center',
+    BOTTOM_RIGHT = 'bottom-right',
+    TOP = 'top',
+    BOTTOM = 'bottom'
+}
+
 interface BasePromptConfig {
     uuid: string;
-    type: string;
+    type: PromptTypeName;
     trigger: PromptTrigger;
-    position: string;
+    position: PromptPosition;
```

Comment:
Not all positions are valid for all prompts, how is this scoped for each prompt type?

Reply:
(might find some mileage in `Pick<>` util type)

Reply:
Argh should be applied to concrete config types - it is for alert but missed alert & banner type decls or chopped when tidying